### DISCUSSION TITLE
Foreign key relation in MySQL gets deleted on saving parent entry

### DISCUSSION THREAD

**Discussion by peterhuppertz** [Budibase/budibase#9066] [Help]
I have two tables in a MySQL DB: 1) questionnaire, 2) questions. Each questionnaire can have many questions. I created a one-to-many relation from 1) to 2). Apart from the questions the questionnaire has some other elements like title, introduction, etc.
When I add questions to the questionnaire I set the questionnaires id as foreign key. This works great and I see it correctly reflected in the database.
However, once I save the questionnaire all foreign key connections to 2) get lost. When I try to add them in the "Save row" command I get the error `row[key].map is not a function`.
How can I preserve the foreign key relation once I save the questionnaire entry?

**Comment by andz-bb** [comment]
Hi @peterhuppertz 

could you send an app export so that I can take a closer look at how your app is currently set up? You can send it to support@budibase.com or directly to me [on our Discord server here.](http://discord.gg/E3YXNAg4m5) Thanks!
